internal/internal_workflow_client.go

```diff
@@ -1661,6 +1656,120 @@ func (w *workflowClientInterceptor) ExecuteWorkflow(
 	}, nil
 }
 
+func (w *workflowClientInterceptor) executeWorkflowWithOperation(
+	ctx context.Context,
+	startRequest *workflowservice.StartWorkflowExecutionRequest,
+	operation StartWorkflowOperation,
+) (*workflowservice.StartWorkflowExecutionResponse, error) {
+	startOp := &workflowservice.ExecuteMultiOperationRequest_Operation{
+		Operation: &workflowservice.ExecuteMultiOperationRequest_Operation_StartWorkflow{
+			StartWorkflow: startRequest,
+		},
+	}
+
+	var withStartOp *workflowservice.ExecuteMultiOperationRequest_Operation
+	switch t := operation.(type) {
+	case *UpdateWorkflowOperation:
+		if err := t.execute(); err != nil {
+			return nil, fmt.Errorf("%w: %w", errInvalidWorkflowOperation, err)
+		}
+
+		updateReq, err := w.createUpdateWorkflowRequest(ctx, t.input)
+		if err != nil {
+			return nil, fmt.Errorf("%w: %w", errInvalidWorkflowOperation, err)
+		}
+		if updateReq.WorkflowExecution.WorkflowId == "" {
+			updateReq.WorkflowExecution.WorkflowId = startRequest.WorkflowId
+		}
+
+		withStartOp = &workflowservice.ExecuteMultiOperationRequest_Operation{
+			Operation: &workflowservice.ExecuteMultiOperationRequest_Operation_UpdateWorkflow{
+				UpdateWorkflow: updateReq,
+			},
+		}
+	default:
+		return nil, fmt.Errorf("unsupported operation request type: %T", t)
+	}
+
+	multiRequest := workflowservice.ExecuteMultiOperationRequest{
+		Namespace: w.client.namespace,
+		Operations: []*workflowservice.ExecuteMultiOperationRequest_Operation{
+			startOp,
+			withStartOp,
+		},
+	}
+	multiResp, err := w.client.workflowService.ExecuteMultiOperation(ctx, &multiRequest)
+
+	var multiErr *serviceerror.MultiOperationExecution
+	if errors.As(err, &multiErr) {
+		if len(multiErr.OperationErrors()) != len(multiRequest.Operations) {
+			panic(fmt.Sprintf("MultiOperation error response from server is incomplete: %v instead of %v",
+				len(multiErr.OperationErrors()), len(multiRequest.Operations)))
```

Comment:
Not sure this should be a panic. I get that it's an invariant the server is supposed to uphold, but from the client's perspective it's just bad data returned from the server. Maybe return an error here (with a bit more explanation that server is misbehaving). Same with the following ones.

Reply:
+1 to being an error. I think the SDK should handle bad data from the server as gracefully as it can